pom.xml

```diff
@@ -1014,58 +986,64 @@
                                     not applicable to us. See also
-                                    https://github.com/google/error-prone/issues/488.
-                                    --> -Xep:StaticOrDefaultInterfaceMethod:OFF <!--
-                                    --> -XepOpt:NullAway:AnnotatedPackages=tech.picnic <!--
-                                    --> -XepOpt:NullAway:AssertsEnabled=true <!--
-                                    Append additional custom arguments.
-                                    --> ${error-prone.args}
-                                </arg>
-                                <?SORTPOM RESUME?>
+                                    https://github.com/google/error-prone/issues/488. -->
+                                <arg>-Xep:StaticOrDefaultInterfaceMethod:OFF</arg>
+                                <arg>-XepOpt:NullAway:AnnotatedPackages=tech.picnic</arg>
+                                <arg>-XepOpt:NullAway:AssertsEnabled=true</arg>
+                                <!-- Append additional custom arguments. -->
+                                <arg>${error-prone.args}</arg>
                                 <!-- The Error Prone plugin makes certain
                                 assumptions about the state of the AST at the
                                 moment it is invoked. Those assumptions are met
                                 when using the `simple` compile policy. This
                                 flag may be dropped after resolution of
                                 https://bugs.openjdk.java.net/browse/JDK-8155674. -->
                                 <arg>-XDcompilePolicy=simple</arg>
+                                <!-- Required to work with JDK 16+;
+                                see also https://errorprone.info/docs/installation#jdk-16. -->
+                                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED</arg>
+                                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED</arg>
+                                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED</arg>
+                                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED</arg>
+                                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED</arg>
+                                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED</arg>
+                                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED</arg>
+                                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED</arg>
+                                <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED</arg>
+                                <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED</arg>
```

Comment:
We should not do this, but instead define a `.mvn/jvm.config` file; that yields more performant builds (#2786) and also avoids issues such as #106. More generally, we should port changes from internal builds that are already compatible with JDK 17; that could has already been carefully reviewed and tested.

Reply:
Fixed the link ;)